jib-core/src/main/java/com/google/cloud/tools/jib/ncache/Cache.java

```diff
@@ -57,31 +57,32 @@ private Cache(CacheStorage cacheStorage) {
   }
 
   /**
-   * Saves a cache entry with only a layer {@link Blob}. Use {@link #write(Blob, ImmutableList)} to
-   * include a selector and metadata.
+   * Saves a cache entry with a compressed layer {@link Blob}. Use {@link #write(Blob,
+   * ImmutableList)} to save a cache entry with an uncompressed layer {@link Blob} and include a
+   * selector and metadata.
    *
-   * @param layerBlob the layer {@link Blob}
+   * @param compressedLayerBlob the compressed layer {@link Blob}
    * @return the {@link CacheEntry} for the written layer
    * @throws IOException if an I/O exception occurs
    */
-  public CacheEntry write(Blob layerBlob) throws IOException {
-    return cacheStorage.write(CacheWrite.layerOnly(layerBlob));
+  public CacheEntry write(Blob compressedLayerBlob) throws IOException {
+    return cacheStorage.write(compressedLayerBlob);
   }
 
   /**
-   * Saves a cache entry with a layer {@link Blob}, an additional selector digest, and a metadata
-   * {@link Blob}. Use {@link #write(Blob)} to save only a layer {@link Blob}.
+   * Saves a cache entry with an uncompressed layer {@link Blob}, an additional selector digest, and
+   * a metadata {@link Blob}. Use {@link #write(Blob)} to save a compressed layer {@link Blob}.
    *
-   * @param layerBlob the layer {@link Blob}
+   * @param uncompressedLayerBlob the layer {@link Blob}
    * @param layerEntries the layer entries that make up the layer
    * @return the {@link CacheEntry} for the written layer and metadata
    * @throws IOException if an I/O exception occurs
    */
-  public CacheEntry write(Blob layerBlob, ImmutableList<LayerEntry> layerEntries)
+  public CacheEntry write(Blob uncompressedLayerBlob, ImmutableList<LayerEntry> layerEntries)
```

Comment:
I was thinking since the `compressed` or `uncompressed` qualifier only applies to the `Blob`, having it on the method name sounds a bit off since the `layerEntries` is not compressed or uncompressed, but I do agree distinguishing in the method name would be good. Would there be a better naming than `writeCompressed` vs `writeUncompressed`?

Reply:
Maybe something that points out what is being written, like `writeCompressedBlob`/`Layer`/`Entry`. `writeCompressed` and `writeUncompressed` might be good enough, though.